roles/ceph-validate/tasks/check_devices.yml

```diff
@@ -30,9 +30,9 @@
     - osd_scenario == 'non-collocated'
     - item.failed
 
-- name: fail if dedicated_device is not the size length as devices
+- name: fail if  number of dedicated_devices is not equal to number of devices
   fail:
-    msg: "dedicated_device lengtth must be identical to devices's length"
+    msg: "Number of dedicated_devices must be equal to number of devices"
   when:
     - osd_scenario == 'non-collocated'
-    - dedicated_devices|length != devices|length
\ No newline at end of file
+    - dedicated_devices|length != devices|length
```

Comment:
If ``dedicated_devices`` isn't defined will this error? Or will that first condition of ``osd_scenario == 'non-collocated'`` keep that from happening?

Reply:
If dedicated_devices are not defined, previous task fails -  validate dedicated_device is/are actually device(s)

Reply:
I suspected as much. I think you'll need to add a `` - dedicated_devices is defined`` to the ``when`` clauses in all the tasks that use it.

Reply:
That wouldn't work for all cases because the ``lvm`` scenario could have ``devices`` defined and not ``dedicated_devices``. Actually, I had thought that we'd removed all ceph-disk methods of creating OSDs from master. If that's the case we shouldn't need ``dedicated_devices`` at all anymore.

Reply:
> That wouldn't work for all cases because the `lvm` scenario could have `devices` defined and not `dedicated_devices`.

I think it might work as we are invoking `non_collocated_osd_scenario` only when osd scenario is non-collocated -
https://github.com/ceph/ceph-ansible/blob/master/plugins/actions/validate.py#L87-L88

Reply:
> I think the problem will be that even if you change that validator the tasks in `check_devices.yml` will still fail when `osd_scenario` is `lvm` and `dedicated_devices` isn't provided. You need to only run the tasks in `check_devices.yml` that check `dedicated_devices` only when they are defined.
check_devices tasks are not failing as we have included the condition in when - `- osd_scenario == 'non-collocated'`

> I don't believe that `dedicated_devices` is even required for `non-collocated` as `dedicated_devices` will be defaulted to `devices` if it's not given. http://docs.ceph.com/ceph-ansible/master/osds/scenarios.html#osd-scenario-non-collocated

If `dedicated_devices` are not defined, playbook is failing in ceph-osd task `set_fact build final dedicated_devices list` saying `'dedicated_devices_tmp' is undefined` (Maybe a bug) -
https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-osd/tasks/build_devices.yml#L32

When I had tried non-collocated scenario without defining dedicated_devices long ago, yes it was defaulted to collocated scenario.
I think it is nice if we validate `dedicated_devices` as long as we support luminous, please let me know your views.

Reply:
> I suspected as much. I think you'll need to add a ` - dedicated_devices is defined` to the `when` clauses in all the tasks that use it.

Done - 4d17f8b573433d24b00e5d71b6d1ca2306cb416d